mobile-app/app/screens/AppNavigator/screens/Settings/components/CustomUrlInput.tsx

```diff
@@ -0,0 +1,140 @@
+import { View } from "@components";
+import { WalletTextInputV2 } from "@components/WalletTextInputV2";
+import {
+  ThemedIcon,
+  ThemedSectionTitleV2,
+  ThemedTouchableOpacityV2,
+} from "@components/themed";
+import { CustomServiceProviderType } from "@contexts/CustomServiceProvider";
+import { tailwind } from "@tailwind";
+import { translate } from "@translations";
+import { useEffect, useState } from "react";
+
+export interface CustomServiceProvider {
+  type: CustomServiceProviderType;
+  url: string;
+  defaultUrl: string;
+  label: string;
+  helperText: string;
+}
+
+interface Props extends CustomServiceProvider {
+  inputValue: { url: string; isValid: boolean };
+  isDisabled: boolean;
+  activeInput: CustomServiceProviderType | undefined;
+  setActiveInput: (type: CustomServiceProviderType | undefined) => void;
+  setShowActionButtons: (show: boolean) => void;
+  handleUrlInputChange: (
+    type: CustomServiceProviderType,
+    value: string,
+  ) => void;
+}
+
+export function CustomUrlInput({
+  type,
+  url,
+  defaultUrl,
+  label,
+  helperText,
+  inputValue,
+  isDisabled,
+  activeInput,
+  setActiveInput,
+  setShowActionButtons,
+  handleUrlInputChange,
+}: Props): JSX.Element {
+  const [isUnlocked, setIsUnlocked] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>();
+  const [displayTickIcon, setDisplayTickIcon] = useState<boolean>(true);
+
+  // clear input on unlock
+  useEffect(() => {
+    if (isUnlocked && url === defaultUrl) {
+      return handleUrlInputChange(type, "");
+    }
+  }, [isUnlocked]);
+
+  // show err msg when input is invalid
+  useEffect(() => {
+    if (inputValue.isValid || inputValue.url === "") {
+      setErrMsg(undefined);
+    } else {
+      setErrMsg("Invalid URL");
+    }
+  }, [inputValue]);
+
+  // to display tick icon
+  useEffect(() => {
+    if (!isUnlocked && inputValue.isValid) {
+      return setDisplayTickIcon(true);
+    } else if (inputValue.url === "" && !inputValue.isValid) {
+      return setDisplayTickIcon(false);
+    }
+  }, [inputValue]);
+
+  return (
+    <View style={tailwind("pb-4")}>
+      <ThemedSectionTitleV2
+        testID="endpoint_url_title"
+        text={translate("screens/ServiceProviderScreen", label)}
+      />
+      <View style={tailwind("flex flex-row items-baseline w-full")}>
+        <WalletTextInputV2
+          valid={errMsg === undefined}
+          editable={isUnlocked}
+          value={inputValue.url}
+          inputType="default"
+          onChangeText={(_text: string) => handleUrlInputChange(type, _text)}
```

Comment:
```suggestion
          onChangeText={(text: string) => handleUrlInputChange(type, text)}
```
can we change it to this?